src/spikeinterface/core/zarrextractors.py

```diff
@@ -0,0 +1,424 @@
+        """Returns the number of samples in this signal block
+
+        Returns:
+            SampleIndex: Number of samples in the signal block
+        """
+        return self._timeseries.shape[0]
+
+    def get_traces(
+        self,
+        start_frame: Union[int, None] = None,
+        end_frame: Union[int, None] = None,
+        channel_indices: Union[List, None] = None,
+    ) -> np.ndarray:
+        traces = self._timeseries[start_frame:end_frame]
+        if channel_indices is not None:
+            traces = traces[:, channel_indices]
+        return traces
+
+
+class ZarrSortingExtractor(BaseSorting):
+    """
+    SortingExtractor for a zarr format
+
+    Parameters
+    ----------
+    root_path: str or Path
+        Path to the zarr root file
+    storage_options: dict or None
+        Storage options for zarr `store`. E.g., if "s3://" or "gcs://" they can provide authentication methods, etc.
+
+    Returns
+    -------
+    sorting: ZarrSortingExtractor
+        The sorting Extractor
+    """
+
+    extractor_name = "ZarrSorting"
+    installed = HAVE_ZARR  # check at class level if installed or not
+    mode = "file"
+    # error message when not installed
+    installation_mesg = "To use the ZarrSortingExtractor install zarr: \n\n pip install zarr\n\n"
+    name = "zarr"
+
+    def __init__(self, root_path: Union[Path, str], storage_options: dict | None = None):
+        assert self.installed, self.installation_mesg
+
+        if storage_options is None:
+            if isinstance(root_path, str):
+                root_path_init = root_path
+                root_path = Path(root_path)
+            else:
+                root_path_init = str(root_path)
+            root_path_kwarg = str(Path(root_path).absolute())
+        else:
+            root_path_init = root_path
+            root_path_kwarg = root_path_init
+
+        self._root = zarr.open(root_path_init, mode="r", storage_options=storage_options)
+
+        sampling_frequency = self._root.attrs.get("sampling_frequency", None)
+        num_segments = self._root.attrs.get("num_segments", None)
+        assert "unit_ids" in self._root.keys(), "'unit_ids' dataset not found!"
+        unit_ids = self._root["unit_ids"][:]
+
+        assert sampling_frequency is not None, "'sampling_frequency' attiribute not found!"
+        assert num_segments is not None, "'num_segments' attiribute not found!"
+
+        unit_ids = np.array(unit_ids)
+        assert "spikes" in self._root.keys(), "'spikes' dataset not found!"
+        spikes_group = self._root["spikes"]
+        segment_slices_list = spikes_group["segment_slices"][:]
+        segment_slices = [slice(s[0], s[1]) for s in segment_slices_list]
+
+        BaseSorting.__init__(self, sampling_frequency, unit_ids)
+
+        for segment_index in range(num_segments):
+            soring_segment = ZarrSortingSegment(spikes_group, segment_slices[segment_index], unit_ids)
+            self.add_sorting_segment(soring_segment)
```

Comment:
Another option is to create the `self._cached_spike_vector` and then use the `NumpySortingSegment` (maybe rename as `SpikeVectorSortingSegment`, in a different PR)

Reply:
I think it is relevant to do this in this PR.
The actual ZarrSortingSegment will be slow.

Reply:
done